test/state/player_state_test.cpp

```diff
@@ -161,3 +161,102 @@ TEST_F(PlayerStateTest, GetOffsetFromPosition) {
 }
+
+TEST_F(PlayerStateTest, TowerTest) {
+    player_state::Tower tower1(10);
+    player_state::Tower tower2(19);
+    player_state::Tower tower3(tower1);
+    player_state::Tower tower4;
+    tower4.id = 19;
+
+    EXPECT_EQ(tower1.id, 10);
+    EXPECT_EQ(tower2.id, 19);
+    EXPECT_EQ(tower3.id, 10);
+    EXPECT_EQ(tower4.id, 19);
+
+    EXPECT_NE(tower1, tower2);
+    EXPECT_EQ(tower1, tower3);
+    EXPECT_EQ(tower2, tower4);
+
+    // Initial values
+    EXPECT_EQ(tower4.position, DoubleVec2D(0, 0));
+    EXPECT_EQ(tower4.state, player_state::TowerState::IDLE);
+    EXPECT_EQ(tower4.age, 0);
+    EXPECT_EQ(tower4.hp, 100);
+    EXPECT_EQ(tower4.impact_radius, 0);
+    EXPECT_EQ(tower4.blasting, false);
+
+    tower1.blast();
+    EXPECT_EQ(tower1.blasting, true);
+}
+
+TEST_F(PlayerStateTest, BotTest) {
+    // different methods of construction
+    player_state::Bot bot1(11);
+    player_state::Bot bot2(29);
+    bot2.position = {2, 2};
+    player_state::Bot bot3;
+    bot3.id = 35;
+    player_state::Bot bot4(bot2);
+    player_state::Bot bot5(1);
+    bot5.position = {20, 20};
+
+    EXPECT_EQ(bot1.id, 11);
+    EXPECT_EQ(bot1.final_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot1.transform_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot1.transforming, false);
+    EXPECT_EQ(bot1.blasting, false);
+    EXPECT_EQ(bot1.destination, DoubleVec2D::null);
+    EXPECT_EQ(bot1.position, DoubleVec2D(0, 0));
+
+    // == & != operator checks
+    EXPECT_EQ(bot2, bot4);
+    EXPECT_NE(bot1, bot3);
+    EXPECT_NE(bot1, bot2);
+
+    // target_position!=position case
+    bot1.blast({10, 10});
+    EXPECT_EQ(bot1.final_destination, DoubleVec2D(10, 10));
+    EXPECT_EQ(bot1.blasting, false);
+
+    // target_position==postion case
+    EXPECT_EQ(bot2.position, DoubleVec2D(2, 2));
+    bot2.blast({2, 2});
+    EXPECT_EQ(bot2.transforming, false);
+    EXPECT_EQ(bot2.transform_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot2.final_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot2.position, DoubleVec2D(2, 2));
+    EXPECT_EQ(bot2.blasting, true);
+
+    bot1.reset();
+    EXPECT_EQ(bot1.transform_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot1.final_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot1.transforming, false);
+    EXPECT_EQ(bot1.destination, DoubleVec2D::null);
+    EXPECT_EQ(bot1.blasting, false);
+
+    bot2.reset();
+    EXPECT_EQ(bot2.transform_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot2.final_destination, DoubleVec2D::null);
+    EXPECT_EQ(bot2.transforming, false);
+    EXPECT_EQ(bot2.destination, DoubleVec2D::null);
+    EXPECT_EQ(bot2.blasting, false);
+
+    // transforms test
+    bot1.position = {9, 9};
+    bot1.transform({9, 9});
+    EXPECT_EQ(bot1.transforming, true);
+
+    bot2.transform({10, 10});
+    EXPECT_EQ(bot2.transform_destination, DoubleVec2D(10, 10));
+    EXPECT_EQ(bot2.transforming, false);
+
+    bot2.transform();
+    EXPECT_EQ(bot2.transforming, true);
+}
```

Comment:
Do all operations for bot1 first. After that do bot2 and so on. 
Add new ```TEST_F``` for << overloads.